src/v/cluster/cluster_link/frontend.cc

```diff
@@ -320,6 +387,66 @@ errc frontend::validator::validate_mutation(const cluster_link_cmd& cmd) const {
               return errc::does_not_exist;
           }
           return errc::success;
+      },
+      [this](const cluster::cluster_link_add_mirror_topic_cmd& cmd) {
```

Comment:
Just as a note for other reviewers to understand here:
This validation only works because we take a lock when applying local mutations, otherwise it's possible two commands could race.